changelog/fragments/1740772872-mitigate-symlink-following-when-rotating-files.yaml

```diff
@@ -0,0 +1,36 @@
+# Kind can be one of:
+# - breaking-change: a change to previously-documented behavior
+# - deprecation: functionality that is being removed in a later release
+# - bug-fix: fixes a problem in a previous version
+# - enhancement: extends functionality but does not break or fix existing behavior
+# - feature: new functionality
+# - known-issue: problems that we are aware of in a given version
+# - security: impacts on the security of a product or a user’s deployment.
+# - upgrade: important information for someone upgrading from a prior version
+# - other: does not fit into any of the other categories
+kind: security
+
+# Change summary; a 80ish characters long description of the change.
+summary: mitigate symlink following when rotating files
+
+# Long description; in case the summary is not enough to describe the change
+# this field accommodate a description without length limits.
+# NOTE: This field will be rendered only for breaking-change and known-issue kinds at the moment.
+description: |
+  The agent logger rotates files when necessary.  
+  If an attacker manages to create a symbolic link as the destination file beforehand,  
+  it could cause the agent to write to a sensitive or protected file.  
+  To prevent this, the agent now forces a rotation in such scenarios.
```

Comment:
```suggestion
description: |
  The agent logger rotates files when necessary.  
  Elastic Agent now forces a file rotation when unexpectedly writing to a file through a symbolic link.
```

Same as Beats, let's soften this a bit to look less like a vulnerability disclosure, doing that gets handled separately by InfoSec. https://github.com/elastic/beats/pull/42804/files

Reply:
Done, thanks!